src/ContainerIntegration.jsx

```diff
@@ -15,17 +15,57 @@ export const renderContainerPublishedPorts = ports => {
     if (!ports)
         return null;
 
+    const ranges = [];
     const items = [];
+
+    // create port ranges
     Object.entries(ports).forEach(([containerPort, hostBindings]) => {
-        (hostBindings ?? []).forEach(binding => { // not-covered: null was observed in the wild, but unknown how to reproduce
-            items.push(
-                <ListItem key={ containerPort + binding.HostIp + binding.HostPort }>
-                    { binding.HostIp || "0.0.0.0" }:{ binding.HostPort } &rarr; { containerPort }
-                </ListItem>
-            );
+        const [port, proto] = containerPort.split('/');
+        const portNumber = Number(port);
+        // not-covered: null was observed in the wild, but unknown how to reproduce
+        (hostBindings ?? []).forEach(binding => {
+            const lastRange = ranges[ranges.length - 1];
+            const hostIP = binding.HostIp || "0.0.0.0";
+            const hostPort = Number(binding.HostPort);
+
+            let isPortConsecutive = false
+            let isHostPortConsecutive = false
+            let isSameHostIP = false
+            let isSameProtocol = false
+
+            if (lastRange) {
+                isPortConsecutive = portNumber === lastRange.endPort + 1;
+                isHostPortConsecutive = hostPort === lastRange.hostEndPort + 1;
+                isSameHostIP = hostIP === lastRange.hostIp;
+                isSameProtocol = proto === lastRange.protocol;
+            }
```

Comment:
minor nitpick, I'd prefer a newline between this and the following if statement.